de.metas.adempiere.adempiere/base/src/main/java/de/metas/i18n/po/POTrlRepository.java

```diff
@@ -125,16 +127,52 @@ public final boolean insertTranslations(@NonNull final POTrlInfo trlInfo, final
 		final String keyColumn = trlInfo.getKeyColumnName();
 		final StringBuilder sql = new StringBuilder("INSERT INTO ").append(tableName).append("_Trl (AD_Language,").append(keyColumn).append(", ").append(iColumns)
 				.append(" IsTranslated,AD_Client_ID,AD_Org_ID,Created,Createdby,Updated,UpdatedBy) ")
-				.append("SELECT l.AD_Language,t.").append(keyColumn).append(", ").append(sColumns)
+				.append("SELECT l.").append(I_AD_Language.COLUMNNAME_AD_Language).append(", t.").append(keyColumn).append(", ").append(sColumns)
 				.append(" 'N',t.AD_Client_ID,t.AD_Org_ID,t.Created,t.Createdby,t.Updated,t.UpdatedBy ")
 				.append("FROM AD_Language l, ").append(tableName).append(" t ")
-				.append("WHERE l.IsActive='Y' AND l.IsSystemLanguage='Y' AND l.IsBaseLanguage='N' AND t.").append(keyColumn).append("=").append(recordId)
-				.append(" AND NOT EXISTS (SELECT 1 FROM ").append(tableName).append("_Trl tt WHERE tt.AD_Language=l.AD_Language AND tt.").append(keyColumn).append("=t.").append(keyColumn).append(")");
+				.append("WHERE l."
+						+ I_AD_Language.COLUMNNAME_IsActive
+						+ "='Y'");
+		if (I_AD_Element.Table_Name.equals(tableName))
+		{
+			sql.append(getElementLanguageConditions());
+		}
+		else
+		{
+			sql.append(getGeneralLanguageConditions());
+		}
```

Comment:
what's the benefit of having those 2 methods.
I would suggest to inline them here.
More, take a look, the difference between those 2 conditions is that in one case IsBaseLanguage=Y while in the other case IsBaseLanguage=N

Reply:
Done, as discussed